lib/chef/provider/user.rb

```diff
@@ -66,13 +66,18 @@ def load_current_resource
           end
           current_resource.comment(user_info.gecos)
 
-          if new_resource.password && current_resource.password == "x"
-            begin
-              require "shadow"
-            rescue LoadError
-              @shadow_lib_ok = false
-            else
-              shadow_info = Shadow::Passwd.getspnam(new_resource.username)
+          begin
+            require "shadow"
+          rescue LoadError
+            @shadow_lib_ok = false
+          else
+            shadow_info = Shadow::Passwd.getspnam(new_resource.username)
+            current_resource.inactive(shadow_info.sp_inact&.to_i) if new_resource.inactive
+            if new_resource.expire_date
+              expire_date = DateTime.strptime(shadow_info.sp_expire&.to_s, "%d")
+              current_resource.expire_date(expire_date.strftime("%Y-%m-%d"))
+            end
+            if new_resource.password && current_resource.password == "x"
```

Comment:
I think you can avoid the new_resource.inactive/expire_dates here and rely on `should_set?` doing the nil check against the new_resource.  The current_resource should be the whole current state (although checking `== "x"` against the password really is current state of which password field is being used so that is correct).  The existing check around `new_resource.password` is mostly likely redundant and wrong I would think -- or should definitely also be moved later with a `should_set?` check around actual password setting which might be missing.

Reply:
To be honest I don't fully understand how this all works, just trying to repeat existing patterns in code for the 2 additional fields. The new_resource.password check was there prior so I didn't want to break anything by making extra changes that I'm not sure why it was there to begin with. Biggest issue I have is not having the ability to properly test these changes to ensure they work. Or atleast lack of knowledge on how to test. I believe I'm 90% of the way there. Any help you can provide to get this over the finish line would be greatly appreciated. The biggest feature we need is the inactive setting for CIS audit compliance. We have a workaround for now but figured it'd be good to get into the core resource and would be useful for others.